tensorrt_llm/_torch/auto_deploy/transformations/_graph.py

```diff
@@ -182,6 +188,33 @@ def canonicalize_graph(
     # lint the graph
     gm.graph.lint()
 
+    # lint the graph
+    gm.graph.lint()
+
+
+def canonicalize_graph(
+    gm: GraphModule, shape_prop: bool = False, args_static: Optional[Tuple[Any, ...]] = None
+) -> GraphModule:
+    """Canonicalize the graph of the given GraphModule.
+
+    Args:
+        gm: The GraphModule to canonicalize.
+        shape_prop: Whether to run shape propagation. Shape propagation tends to be finicky and
+            slow, so we only run it optionally.
+        args_static: A tuple of static arguments to use for shape propagation. Shape propagation
+            requires all inputs to the graph ("placeholder" nodes) to have metadata with an
+            appropriate FakeTensor argument (``node.meta["val"]``). ``args_static`` can be used to
+            infer static FakeTensor information if some placeholder nodes do not have metadata.
+            When ``meta["val"]`` is available, it will take precedence over ``args_static``.
+
+    Returns:
+        The canonicalized (cleaned-up) GraphModule.
+    """
+    ad_logger.debug(f"Before canonicalizing: {gm}")
+
+    for _, subgm in named_graphmodules(gm):
+        _canonicalize_single_gm(subgm, shape_prop=shape_prop, args_static=args_static)
```

Comment:
`args_static` will only work for the "root" graphmodule. We should not pass them in for sub-graphmodules